pkg/controllers/node/termination/controller.go

```diff
@@ -87,7 +89,11 @@ func (c *Controller) finalize(ctx context.Context, node *corev1.Node) (reconcile
 	if !controllerutil.ContainsFinalizer(node, v1.TerminationFinalizer) {
 		return reconcile.Result{}, nil
 	}
-	nodeClaims, err := nodeutils.GetNodeClaims(ctx, node, c.kubeClient)
+	if !nodeutils.IsManaged(node, c.cloudProvider) {
+		return reconcile.Result{}, nil
+	}
```

Comment:
Does this ever happen because of the event filter?

Reply:
It can, the event filter only applies when enqueuing the reconcile request, it's possible it no longer holds true by the time it's processed.